deltachat-ffi/src/lib.rs

```diff
@@ -123,13 +122,14 @@ pub unsafe extern "C" fn dc_get_config(
     key: *mut libc::c_char,
 ) -> *mut libc::c_char {
     assert!(!context.is_null());
-    assert!(!key.is_null(), "invalid key");
     let context = &*context;
 
-    match config::Config::from_str(dc_tools::as_str(key)) {
-        Ok(key) => context.get_config(key).unwrap_or_default().strdup(),
-        Err(_) => std::ptr::null_mut(),
-    }
+    assert!(!key.is_null(), "invalid key pointer");
+    let key = config::Config::from_str(dc_tools::as_str(key)).expect("invalid key");
+
+    // TODO: Translating None to NULL would be more sensible than translating None
+    // to "", as it is now.
```

Comment:
@r10s can you comment on whether the current behaviour is correct please?  Or should it indeed be NULL?

Reply:
What is correct? I did not change behavior with my patch, but API that deliberately merges missing key and key with value "" is dubious.